### PR TITLE
PP 11681 Replace 'base-client' with 'axios-base-client' in 'connector.client'

### PR DESCRIPTION
**Pull request by JFSGDS**
https://payments-platform.atlassian.net/browse/PP-11681

## WHAT

- Change ‘connector.client’ methods to use ‘axios-base-client’.
- Refactor call parameters initialisation.
- Use ‘async/await’ syntax.

## HOW 
All automated tests should pass without the need to be amended.
Existing functionality should remain unaffected.

### CODE REVIEW

File: app/controllers/transactions/transaction-list.controller.it.test.js [hunk 1]
```diff
@@ -51,9 +51,15 @@ describe('The /transactions endpoint', () => {
       // No mocking defined on purpose to mock a network failure,
       // This integration test will cover server errors outside the 500 and 504 defined in the Cypress test
       await transactionListController(req, res, next)
-      const expectedError = sinon.match.instanceOf(Error)
+      // Temorary replacement for test failing with axios
+      sinon.match.instanceOf(Error)
```
Comment (iqbalgds) [inline app/controllers/transactions/transaction-list.controller.it.test.js:55]: Just need to figure this test out.

Comment (JFSGDS) [reply app/controllers/transactions/transaction-list.controller.it.test.js]: - 'try - catch' clause added in `connector.client.js` in order to throw an error if unable to retrieve transaction or card types.
- associated test reverted to original assertion so, in effect, no amendment needed.

File: app/services/clients/connector.client.js [hunk 2]
```diff
@@ -371,143 +318,111 @@ ConnectorClient.prototype = {
...
-      description: 'Retrieves all card types',
-      service: SERVICE_NAME
-    })
+    configureClient(client, url)
+    try {
+      const response = await client.get(url, 'Retrieves all card types')
+      return response.data
+    } catch (err) {
+      throw new Error('Unable to retrieve list of transactions or card types.')
+    }
```
Comment (iqbalgds) [inline app/services/clients/connector.client.js:345]: Can we get rid of this try/catch?
So it's consistent with the others?

Comment (iqbalgds) [reply app/services/clients/connector.client.js:345]: Alternatively, we may need to update the test that may be connected to this.

File: app/services/clients/connector.client.js [hunk 3]
```diff
@@ -289,80 +262,54 @@ ConnectorClient.prototype = {
...
-        json: true,
-        body: params.payload,
-        description: 'Check Worldpay 3DS Flex credentials',
-        service: SERVICE_NAME
-      }
-    )
+  postCheckWorldpay3dsFlexCredentials: async function (params) {
+    const url = `${this.connectorUrl}${CHECK_WORLDPAY_3DS_FLEX_CREDENTIALS_PATH.replace('{accountId}', params.gatewayAccountId)}`
+    configureClient(client, url)
+    const response = await client.post(url, params.payload, 'patch gateway account credentials')
```
Comment (iqbalgds) [inline app/services/clients/connector.client.js:268]: Please check the Description here

**Review by iqbalgds — CHANGES_REQUESTED** [review]
Looks really good, some minor comments

File: app/services/clients/connector.client.js (hunk 3 above)
Comment (JFSGDS) [reply app/services/clients/connector.client.js]: Changed as requested.

File: app/services/clients/connector.client.js [hunk 4]
```diff
@@ -516,181 +427,120 @@ ConnectorClient.prototype = {
...
-      baseUrl: this.connectorUrl,
-      url,
-      body: payload,
-      description: 'switch account payment service provider',
-      service: SERVICE_NAME
-    })
+  postAccountSwitchPSP: async function (gatewayAccountId, payload) {
+    const url = `${this.connectorUrl}${SWITCH_PSP_PATH.replace('{accountId}', gatewayAccountId)}`
+    configureClient(client, url)
+    const response = await client.post(url, payload, 'get a charge')
```
Comment (iqbalgds) [inline app/services/clients/connector.client.js:542]: Description needs to be updated to match

**Review by iqbalgds — CHANGES_REQUESTED** [review]
Looks good - tiny update.

File: app/services/clients/connector.client.js (hunk 4 above)
Comment (JFSGDS) [reply app/services/clients/connector.client.js]: Changed as requested. Also spotted another i.e. `updateConfirmationEmailEnabled` description s/b `'update confirmation email enabled'`

**Review by iqbalgds — APPROVED** [review]
Looks great - just need to squash commits

**Pull request merged**